frontend/DataView/DataView.js

```diff
@@ -74,9 +74,20 @@ class DataView extends React.Component<DataViewProps> {
       return <div style={missingStyle(theme)}>null</div>;
     }
 
+    const dataType = typeof data;
+    var isSingleValue = dataType === 'number' || dataType === 'string' || dataType === 'boolean';
     var isArray = Array.isArray(data);
     var elements = [];
-    if (isArray) {
+    if (isSingleValue) {
+      elements.push(<DataItem
+          name={''}
+          hideName={true}
+          path={this.props.path}
+          inspect={this.props.inspect}
+          showMenu={this.props.showMenu}
+          readOnly={this.props.readOnly}
```

Comment:
This case should be read-only, because we can't actually edit the context value without crawling upwards and mutating the ancestor context. (We _can_ override the value on the instance, but React will reset it when we call `forceUpdate` so the change won't actually stick.)